### PR TITLE
Add Theme Configuration options

### PR DESCRIPTION
**Pull request by cool-pants**
# Feature
As mentioned in #28, add theme configuration functionality. A sample setup for tokyonight is provided below:
```lua
--- [in my nvim config dir] lua/modules/themes/tokyonight.lua
---@type Huez.ThemeConfig
---@diagnostic disable-next-line: missing-fields
local M = {
	styles = { "moon", "day", "night", "storm" },
}

---@param theme string
M.set_theme = function(theme)
	---@type tokyonight.Config
	local base_conf = {
		styles = {
			functions = { bold = true, italic = true },
			variables = { italic = true },
			keywords = { bold = true, italic = true },
		},
		on_colors = function(colors) end,
		on_highlights = function(highlights, colors)
			highlights["Type"] = { bold = true, italic = true }
			highlights["@string.javascript"] = { bold = true, italic = true, fg = colors.green }
		end,
	}

	require("tokyonight").setup(base_conf)
	vim.cmd("colorscheme " .. theme)
	return true
end

return M
```

To load the theme, we just add one more parameter to the `huez.setup` command.:
```lua
require("huez").setup({
	theme_config_module = "modules.themes",
})
```
PS: If no `theme_config_module` is given, the themes will load as per the default `vim.cmd.colorscheme`

## In Action
[huez-doc.webm](https://github.com/vague2k/huez.nvim/assets/60956127/10a4162c-a34e-427b-b72f-151c1b06b0ab)

### CODE REVIEW

**Comment by vague2k** [comment]
Thanks for the PR! I'll look at it and make changes if neccessary, will probably get merged today

File: README.md [hunk 1]
```diff
@@ -75,6 +78,109 @@ Huez comes with the following defaults
 }
 ```
 
+### Extending theme configuration
```
Comment (vague2k) [inline README.md:81]: This will probably get moved to the wiki

File: lua/huez/config.lua [hunk 2]
```diff
@@ -88,6 +105,72 @@ M.set = function(user_opts)
...
+
+-- Will load all the theme configs into memory.
+M.load_theme_configs = function()
+  local theme_config_module = M.current.theme_config_module
+
+  local config_path = vim.fn.stdpath("config")
+
+  -- Convert module string to dir path string to check if dir exists
+  -- TODO: find a better way to check this
+  local theme_config_dir = config_path .. "/lua/" .. vim.fn.substitute(theme_config_module, ".", "/", "g")
```
Comment (vague2k) [inline lua/huez/config.lua:119]: This is fine for now, the todo comment can be removed

File: lua/huez/config.lua [hunk 3]
```diff
@@ -88,6 +105,72 @@ M.set = function(user_opts)
...
+    ok, conf = pcall(require, theme_config_module .. "." .. theme)
+    -- if cannot load config, means no config found => continue
+    if not ok then
+      goto continue
+    end
+
+    logger("found configs for " .. theme, "INFO")
+    for _, theme_style in pairs(conf.styles) do
+      logger("Loading config for " .. theme .. " " .. theme_style, "INFO")
+      M.theme_setters[theme .. "-" .. theme_style] = conf.set_theme
```
Comment (vague2k) [inline lua/huez/config.lua:149]: this assumes all themes a user has installed will have a `theme-style` in the namespace. Some themes like neomodern or base16 bundle many themes together that are seperate themes not denoted by a dash.

for example there's `neomodern` and `rose-prime` which are both themes contained within the same colorscheme plugin. This probably needs to be changed/handled differently

**Review by vague2k — CHANGES_REQUESTED** [review]
See comments in the config file, I'm actively trying to work on it but it's a bit tricky 

edit: if you have any other behavior details im missing out on, please let me know !

File: lua/huez/config.lua (hunk 3 above)
Comment (cool-pants) [reply lua/huez/config.lua:160]: Yeah just looked into this. Neomodern does have a different way of handling these issues. I made a fix making the `styles` component as optional and adding a default configuration entry for each theme which is the theme name itself.

**Comment by cool-pants** [comment]
No issues. I just noticed another thing, which seems a bit weird and maybe a behaviour of the theme instead of `huez`.

Setting the theme to certain themes, messes up lualine default theme upon startup. Currently noticing this for `neomodern` themes.

[Possible bug?](https://github.com/vague2k/huez.nvim/assets/60956127/0e91529b-b4c7-42ca-8d73-3885031f8e3e)

**Comment by vague2k** [comment]
> No issues. I just noticed another thing, which seems a bit weird and maybe a behaviour of the theme instead of `huez`.
> 
> Setting the theme to certain themes, messes up lualine default theme upon startup. Currently noticing this for `neomodern` themes.
> 
>  Screencast.from.07-07-24.09.05.02.PM.IST.webm

this seems like an issue with the theme rather than huez, i notice the sort of issue sometimes with tokyonight aswell

**Comment by vague2k** [comment]
currently testing, but for some reason it's saying i don't have a setter for any of my themes, even though I have a setter for rose-prime, weird. 

I reviewed the code and it makes sense so far, not sure why it works for you and not me.
Screenshot 2024-07-07 at 12 47 59 PM

**Comment by vague2k** [comment]
ahhh i see why i think gimme a sec ill review it

**Review by vague2k — APPROVED** [review]
Tested on my end, everything looks good. 

Ignore the format tests failing, for some reason stylua doesn't recognize the `goto` keyword and this is a bug with stylua last reported as of a couple days ago.

**Pull request merged**